xbmc/addons/AddonManager.h

```diff
@@ -331,6 +331,38 @@ namespace ADDON
 
     AddonOriginType GetAddonOriginType(const AddonPtr& addon) const;
 
+    /*!
+     * @brief Addon update and install managment.
+     *
+     * Parts inside here are used for changes about addon system.
+     *
+     * @warning This should be never used from other places outside of addon
+     * system directory.
+     *
+     * Currently listed call sources:
+     * - @ref CAddonInstallJob::DoWork
+     */
+    /*@{{{*/
+
+    /*!
+     * @brief Update addon origin data.
+     *
+     * This becomes called from @ref CAddonInstallJob to set the source repo and
+     * if update, to set also the date about.
+     *
+     * @note This must be called after the addon manager has with @ref FindAddons()
+     * inserted a new addon inside database.
```

Comment:
`...after the addon manager has inserted a new addon with @ref FindAddons() into database.`